libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/GovCert.hs

```diff
@@ -15,34 +15,50 @@
 
 module Test.Cardano.Ledger.Conformance.ExecSpecRule.Conway.GovCert (nameGovCert) where
 
-import Cardano.Ledger.BaseTypes (Inject)
+import Cardano.Ledger.Address (RewardAccount)
+import Cardano.Ledger.BaseTypes (Inject (..))
 import Cardano.Ledger.Coin (Coin)
 import Cardano.Ledger.Conway
+import Cardano.Ledger.Conway.Governance (VotingProcedures)
 import Cardano.Ledger.Conway.TxCert
 import Cardano.Ledger.Crypto (StandardCrypto)
-import Constrained (lit)
+import Constrained
 import Data.Bifunctor (Bifunctor (..))
 import qualified Data.List.NonEmpty as NE
 import Data.Map.Strict (Map)
 import qualified Data.Text as T
 import qualified Lib as Agda
 import Test.Cardano.Ledger.Conformance
-import Test.Cardano.Ledger.Conformance.ExecSpecRule.Conway.Base
+import Test.Cardano.Ledger.Conformance.ExecSpecRule.Conway.Base (
+  ConwayCertExecContext (..),
+  conwayCertExecContextSpec,
+ )
 import Test.Cardano.Ledger.Constrained.Conway
+import Test.Cardano.Ledger.Constrained.Conway.WitnessUniverse
+
+instance Inject (WitUniv Conway, ConwayCertExecContext Conway) (Map (RewardAccount StandardCrypto) Coin) where
+  inject (_, x) = ccecWithdrawals x
+
+instance Inject (WitUniv Conway, ConwayCertExecContext Conway) (VotingProcedures Conway) where
+  inject (_, x) = ccecVotes x
 
 instance
-  ( IsConwayUniv fn
-  , Inject (ConwayCertExecContext Conway) (Map (DepositPurpose StandardCrypto) Coin)
-  ) =>
+  IsConwayUniv fn =>
   ExecSpecRule fn "GOVCERT" Conway
   where
-  type ExecContext fn "GOVCERT" Conway = ConwayCertExecContext Conway
+  type ExecContext fn "GOVCERT" Conway = (WitUniv Conway, ConwayCertExecContext Conway)
```

Comment:
It'd be better to add a witness universe field to `ConwayCertExecContext` and then get rid of the pair here

Reply:
same as before